File: app/src/main/java/com/onix/streamer/demo/actioncamera/binding/CamStateObservable.java
package com.onix.streamer.demo.actioncamera.binding;

import android.databinding.ObservableInt;
import android.view.View;

import com.onix.avlib.ActionStreamerState;

import java.util.Arrays;
import java.util.List;

public class CamStateObservable extends ObservableInt {
    private List<ActionStreamerState> mEvents;

    public CamStateObservable(ActionStreamerState... events) {
        mEvents = Arrays.asList(events);
        set(View.INVISIBLE);
    }

    public void set(ActionStreamerState event) {
        set(mEvents.contains(event) ? View.VISIBLE : View.INVISIBLE);
    }
}

File: app/src/main/java/com/onix/avlib/demo/action_camera/CamStateObservable.kt
package com.onix.avlib.demo.action_camera

import android.view.View
import androidx.databinding.ObservableInt
import com.onix.avlib.ActionStreamerState

class CamStateObservable(vararg events: ActionStreamerState) : ObservableInt() {
    private val mEvents: List<ActionStreamerState> = listOf(*events)

    init {
        set(View.INVISIBLE)
    }

    fun set(event: ActionStreamerState) {
        set(if (mEvents.contains(event)) View.VISIBLE else View.INVISIBLE)
    }
}